satpy/readers/seviri_base.py

```diff
@@ -910,3 +910,34 @@ def pad_data_vertically(data, final_size, south_bound, north_bound):
     padding_north = get_padding_area(((final_size[0] - north_bound), ncols), data.dtype)
 
     return np.vstack((padding_south, data, padding_north))
+
+
+def mask_bad_quality(data, line_validity, line_geometric_quality, line_radiometric_quality):
```

Comment:
To improve readability this method could be split into smaller methods like so

```
def mask_bad_quality(data, line_flags):
    line_mask = _get_bad_quality_lines(*line_flags)
    return _mask_bad_quality_lines(data, line_mask)
```

Even if `_mask_bad_quality_lines` is a one-liner, it helps keeping the contents of `mask_bad_quality` at the same level of abstraction.

Reply:
I am not sure if this would improved readability since it fragments the code into really small pieces. I mean the additional functions that would be created would have no real functional purpose on their own. But if this is wanted I can refactor.

Reply:
Yes, this leads to many small pieces. But if named properly they document the behavior very well. You can read the code like a news article from top to bottom with increasing level of detail. It should be possible for readers to quickly grasp what the method is doing and then decide whether they want to know the details.

Note that I'm just reading the "Clean Code" book by Robert C. Martin and are very enthusiastic  about it :)

Reply:
I'm not against what @sfinkens has said, but I don't completely agree either. The result he posted in his original comment is indeed easier to read, but I'm not sure it is necessary based on the level of abstraction/operations being done in the function. Yes, what he suggested would be considered best practice as far as "Clean Code" goes, but this might be a case where it isn't a big deal to disobey it. I could also see an argument for combining the line_mask lines into one or two lines (combining the `&`s) or renaming this method to `mask_bad_quality_lines` so it is more clear it is working with lines. The last thing is that the `np.choose` line is ugly (because it has to be) and with a lot of arguments that aren't arguments to the function (np.nan, np.newaxis, etc) so it feels like it should be refactored.

My final opinion is to not refactor it. Leave it as is. But I'd like to hear other opinions.

Reply:
I'm with @sfinkens here.

Reply:
I refactored the creation of the scan lines mask into a separate function. Hope this is how you wanted it to look like.